libminifi/src/core/ProcessSession.cpp

```diff
@@ -537,63 +534,82 @@ void ProcessSession::import(std::string source, const std::shared_ptr<core::Flow
   }
 }
 
-void ProcessSession::import(std::string source, std::vector<std::shared_ptr<FlowFileRecord>> &flows, bool keepSource, uint64_t offset, char inputDelimiter) {
+void ProcessSession::import(const std::string& source, std::vector<std::shared_ptr<FlowFileRecord>> &flows, uint64_t offset, char inputDelimiter) {
   std::shared_ptr<ResourceClaim> claim;
+  std::shared_ptr<io::BaseStream> stream;
   std::shared_ptr<FlowFileRecord> flowFile;
-  int size = getpagesize();
-  std::vector<char> charBuffer;
-  charBuffer.resize(size);
 
+  std::vector<uint8_t> buffer(getpagesize());
   try {
-    // Open the input file and seek to the appropriate location.
-    std::ifstream input;
-    logger_->log_debug("Opening %s", source);
-    input.open(source.c_str(), std::fstream::in | std::fstream::binary);
-    if (input.is_open() && input.good()) {
-      if (offset != 0) {
+    try {
+      std::ifstream input;
```

Comment:
is this a change in the spacing? Did you apply the linter?

Reply:
GitHub's diff is really hard to read properly, but this is not a spacing change, it is a new try-catch block.
Along with the change in the catch blocks, this is to eliminate a code duplication. (We log the errors separately for std::exception and other exceptions, then rethrow the exception and decrement the claim refcounter).

Reply:
Ah, so the behavior of decrementing ( I think I have a comment elsewhere ) is maintained? if so you can ignore that comment. Thanks!